### PR TITLE
[JENKINS-55234] Update parent pom to benefit from new Mockito version

### PR DESCRIPTION
**Pull request by alecharp**
[Issue](https://issues.jenkins-ci.org/browse/JENKINS-55234)

Updating the parent pom so we can benefit from the Mockito Managed version.

@jenkinsci/java11-support

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
JEP-200 in tests :(

**Comment by carlossg** [comment]
This unmarshalling error only happens some times and only in windows, what could be the issue?

ie.
https://ci.jenkins.io/job/Plugins/job/kubernetes-plugin/job/PR-409/1/testReport/org.csanchez.jenkins.plugins.kubernetes/KubernetesSlaveTest/linux_8___Archive__linux_8____testGetPodRetention/
https://ci.jenkins.io/job/Plugins/job/kubernetes-plugin/job/PR-409/1/testReport/org.csanchez.jenkins.plugins.kubernetes/KubernetesSlaveTest/windows_8___Archive__windows_8____testGetPodRetention/

**Comment by oleg-nenashev** [comment]
`Refusing to marshal org.jvnet.hudson.test.TestCrumbIssuer for security reasons` .
It seems it needs a fix in JTH: https://github.com/jenkinsci/jenkins-test-harness/blob/53f82026d9d143535dd693c2617fd449c3a49741/src/main/java/org/jvnet/hudson/test/TestCrumbIssuer.java

**Comment by carlossg** [comment]
that wouldn't explain why it fails only on windows, would it?

**Comment by carlossg** [comment]
filed as https://issues.jenkins-ci.org/browse/JENKINS-55240

File: pom.xml [hunk 1]
```diff
@@ -287,6 +286,11 @@
         <artifactId>junit</artifactId>
         <version>1.3</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-lang3</artifactId>
+        <version>3.7</version>
+      </dependency>
```
Comment (batmat) [inline pom.xml]: That addition alone looks very fishy. Did you mean to add it with `test` scope?

Comment (carlossg) [reply pom.xml]: it is needed for enforcer.RequireUpperBoundDeps

**Pull request merged**